src/ranged.cpp

```diff
@@ -567,6 +567,10 @@ int player::fire_gun( const tripoint &target, int shots, item& gun )
     moves -= time_to_fire( *this, *gun.type );
 
     practice( skill_gun, xp * ( get_skill_level( skill_gun ) + 1 ) );
+    if( !xp && one_in( 10 ) ) {
+        add_msg_if_player( m_info, _( "You'll need a more accurate gun to keep improving your aim." ) );
```

Comment:
We'll need a more accurate message.

It needs to cover marksmanship and distance to target.